src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterPreTFMStep.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
+{
+    /// <summary>
+    /// Upgrade step that updates NuGet package references
+    /// to better work after upgrade. Packages references are
+    /// updated if the reference appears to be transitive (with
+    /// SDK style projects, only top-level dependencies are necessary
+    /// in the project file), if the package version doesn't
+    /// target a compatible .NET framework but a newer version does,
+    /// or if the package is explicitly mapped to an updated
+    /// NuGet package in a mapping configuration file.
+    /// </summary>
+    public class PackageUpdaterPreTFMStep : UpgradeStep
```

Comment:
Do we need a full copy like this? Can't we just subclass the current package updater? Or create a common base class?

Reply:
+1

Reply:
Changed